ui/app/components/raft-storage-overview.js

```diff
@@ -0,0 +1,67 @@
+import Component from '@ember/component';
+import { getOwner } from '@ember/application';
+import config from '../config/environment';
+import { inject as service } from '@ember/service';
+
+export default Component.extend({
+  flashMessages: service(),
+  useServiceWorker: null,
+
+  async init() {
+    this._super(...arguments);
+    if (this.useServiceWorker === false) {
+      return;
+    }
+    if ('serviceWorker' in navigator) {
+      let worker = await navigator.serviceWorker.getRegistration(config.serviceWorkerScope);
+      if (worker) {
+        this.set('useServiceWorker', true);
+      }
+    }
+  },
+
+  actions: {
+    async removePeer(model) {
+      let { nodeId } = model;
+      try {
+        await model.destroyRecord();
+      } catch (e) {
+        let errString = e.errors ? e.errors.join(' ') : e.message || e;
+        this.flashMessages.danger(`There was an issue removing the peer ${nodeId}: ${errString}`);
+        return;
+      }
+      this.flashMessages.success(`Successfully removed the peer: ${nodeId}.`);
+    },
+
+    downloadViaServiceWorker() {
+      this.flashMessages.success('The snapshot download will begin shortly.');
+    },
+
+    async downloadSnapshot() {
+      let adapter = getOwner(this).lookup('adapter:application');
+
+      this.flashMessages.success('The snapshot download has begun.');
+      let resp, blob;
+      try {
+        resp = await adapter.rawRequest('/v1/sys/storage/raft/snapshot', 'GET');
+        blob = await resp.blob();
+      } catch (e) {
+        let errString = e.errors ? e.errors.join(' ') : e.message || e;
+        this.flashMessages.danger(`There was an error trying to download the snapshot: ${errString}`);
+      }
+      let filename = 'snapshot.gz';
+      let file = new Blob([blob], { type: 'application/x-gzip' });
+      file.name = filename;
+      if ('msSaveOrOpenBlob' in navigator) {
+        navigator.msSaveOrOpenBlob(file, filename);
```

Comment:
i went to google this since i've never read up on `navigator` or `msSaveOrOpenBlob` and noticed that it's [obsolete/nonstandard](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/msSaveOrOpenBlob). what's it used for? is this the mechanism which actually downloads the file?

Reply:
This is for IE11 support 😭. So the download will happen with the service worker if it's supported, and if not, this is the backup mechanism.